dags/glam.py

```diff
@@ -0,0 +1,157 @@
+    depends_on_past=True,
+    date_partition_parameter=None,
+    arguments=('--replace',),
+    dag=dag)
+
+clients_daily_histogram_aggregates = bigquery_etl_query(
+    task_id="clients_daily_histogram_aggregates",
+    destination_table="clients_daily_histogram_aggregates_v1",
+    dataset_id=dataset_id,
+    project_id="moz-fx-data-shared-prod",
+    owner="msamuel@mozilla.com",
+    email=["telemetry-alerts@mozilla.com", "msamuel@mozilla.com"],
+    depends_on_past=False,
+    dag=dag)
+
+sql_file_path = "sql/{}/{}/query.sql".format(dataset_id, "clients_daily_keyed_histogram_aggregates_v1")
+clients_daily_keyed_histogram_aggregates = bigquery_etl_query(
+    task_id="clients_daily_keyed_histogram_aggregates",
+    destination_table="clients_daily_histogram_aggregates_v1",
+    sql_file_path=sql_file_path,
+    dataset_id=dataset_id,
+    project_id="moz-fx-data-shared-prod",
+    owner="msamuel@mozilla.com",
+    email=["telemetry-alerts@mozilla.com", "msamuel@mozilla.com"],
+    depends_on_past=True,
+    arguments=('--append_table','--noreplace',),
+    dag=dag)
+
+clients_histogram_aggregates = bigquery_etl_query(
+    task_id="clients_histogram_aggregates",
+    destination_table="clients_histogram_aggregates_v1",
+    sql_file_path=sql_file_path,
+    dataset_id=dataset_id,
+    project_id="moz-fx-data-shared-prod",
+    owner="msamuel@mozilla.com",
+    email=["telemetry-alerts@mozilla.com", "msamuel@mozilla.com"],
+    depends_on_past=True,
+    date_partition_parameter=None,
+    arguments=('--replace',),
+    dag=dag)
+
+sql_file_path = "sql/{}/{}/query.sql".format(dataset_id, "clients_scalar_probe_counts_v1")
+client_scalar_probe_counts = bigquery_etl_query(
+    task_id="client_scalar_probe_counts",
+    destination_table="client_probe_counts_v1",
+    sql_file_path=sql_file_path,
+    dataset_id=dataset_id,
+    project_id="moz-fx-data-shared-prod",
+    owner="msamuel@mozilla.com",
+    email=["telemetry-alerts@mozilla.com", "msamuel@mozilla.com"],
+    depends_on_past=True,
+    gke_cluster_name="test",
+    date_partition_parameter=None,
+    dag=dag)
+
+sql_file_path = "sql/{}/{}/query.sql".format(dataset_id, "clients_histogram_probe_counts_v1")
+client_histogram_probe_counts = bigquery_etl_query(
+    task_id="client_histogram_probe_counts",
+    destination_table="client_probe_counts_v1",
+    sql_file_path=sql_file_path,
+    dataset_id=dataset_id,
+    project_id="moz-fx-data-shared-prod",
+    owner="msamuel@mozilla.com",
+    email=["telemetry-alerts@mozilla.com", "msamuel@mozilla.com"],
+    depends_on_past=True,
+    date_partition_parameter=None,
+    arguments=('--append_table','--noreplace',),
+    dag=dag)
+
+
+wait_for_main_ping >> clients_daily_scalar_aggregates
+clients_daily_scalar_aggregates >> clients_daily_keyed_scalar_aggregates
+clients_daily_keyed_scalar_aggregates >> clients_daily_keyed_boolean_aggregates
+clients_daily_keyed_boolean_aggregates >> clients_scalar_aggregates
```

Comment:
Can these three last steps all run in parallel? If so, you can define them all as dependent on `clients_daily_scalar_aggregates` instead of a chain. That will allow it to run a bit faster.

Reply:
I think we can do:

```
clients_daily_scalar_aggregates >> clients_daily_keyed_scalar_aggregates
clients_daily_scalar_aggregates >> clients_daily_keyed_boolean_aggregates
clients_daily_keyed_boolean_aggregates >> clients_scalar_aggregates
clients_daily_keyed_scalar_aggregates >> clients_scalar_aggregates
```

`clients_daily_scalar_aggregates` should run first to replace a partition. `clients_daily_keyed_scalar_aggregates` and `clients_daily_keyed_boolean_aggregates` can run in parallel afterwards, and `clients_scalar_aggregates` needs to wait for all the data to get into the table its querying from so it needs to wait on those two tasks.